### PR TITLE
Add Clarity, Rholang, and Vyper

### PR DESCRIPTION
**Pull request by chiedo**
## Checklist:

- [x] **I am adding a new language.**
  - [x] The extension of the new language is used in hundreds of repositories on GitHub.com.
    - Search results for each extension:
      
      - https://github.com/search?q=extension%3Avy&type=Code
      - https://github.com/search?q=extension%3Arho&type=Code
  - [x] I have included a real-world usage sample for all extensions added in this PR:
    - [x] https://github.com/rchain/Rholang/blob/master/examples/regular/token.rho
    - [x] https://github.com/t4sk/vyper-by-example/edit/master/src/Uniswap.vy
    - [x] https://github.com/rchain/Rholang/blob/master/LICENSE
    - [x] https://github.com/t4sk/vyper-by-example/blob/master/LICENSE
  - [x] I have included a syntax highlighting grammar
  - [x] I have updated the heuristics to distinguish my language from others using the same extension.

### CODE REVIEW

**Comment by chiedo** [comment]
Not sure what's expected here!

`I have updated the heuristics to distinguish my language from others using the same extension.`

But think I took care of everything else!

**Review by lildude — CHANGES_REQUESTED** [review]
A heuristic is only needed when there is a common extension that is shared between multiple languages, so if these are all unique, you don't need to worry about it.

Please replace all the "hello worlds" with real-world examples. These are used to train the classifier. Replace reference the source of the samples and their respective licenses in the OP too.

As for Clarity support... I've pointed out that the grammar you've selected is license under GPL. We can't ship this. That said, there's already a PR open for Clarity at https://github.com/github/linguist/pull/5665 so lets rely on that one and use this just for the other two languages.

File: lib/linguist/languages.yml
Comment (chiedo) [reply lib/linguist/languages.yml]: Hmmm.. Things didn't seem alphabetical but I just gave it a shot and added those new entries to the places that seemed correct!

File: vendor/licenses/git_submodule/clarity-lsp.dep.yml
Comment (chiedo) [reply vendor/licenses/git_submodule/clarity-lsp.dep.yml]: I removed Clarity!

Comment (chiedo) [reply vendor/licenses/git_submodule/clarity-lsp.dep.yml]: And added new examples. How does the PR and OP look now?

**Comment by davidhq** [comment]
My repo which is already used for Solidity has Vyper syntax 
https://github.com/davidhq/SublimeEthereum/tree/linguist

**Comment by davidhq** [comment]
hmm seems that `.vy` extension is already working?

Screenshot 2022-02-03 at 19 39 39

**Comment by davidhq** [comment]
Here was discussion in 2019: https://github.com/github/linguist/pull/4453

not sure what happened and if vyper was added shortly afterwards ?

It does seem to be working and syntax is visible on `.py` files, however it's not listed here https://github.com/github/linguist/tree/master/vendor, just in [grammars.yml](https://github.com/github/linguist/blob/master/grammars.yml#L122) as mentioned.

**Comment by davidhq** [comment]
So it would be good to double check ... :)

I took a look at https://github.com/tintinweb/vscode-vyper/blob/master/syntaxes/vyper.tmLanguage.json .. and it actually doesn't have any vyper native stuff... it's just straight python syntax.

If there is clarification that my syntax is currently used for `.vy` files and will remain so, I can invest more time in bringing it to same quality as latest Solidity syntax in the same repo is (more or less perfect).

**Comment by chiedo** [comment]
It looks like there's also a PR already for Clarity! https://github.com/github/linguist/pull/5665

Given that there's Vyper and Clarity are both already covered, I think I'll go ahead and close this!

**Comment by davidhq** [comment]
@chiedo this is still useful, isn't it?

https://github.com/github/linguist/pull/5749/commits/7b98773827b87f61fe3cb1ba3db3f955735aebfa#diff-9c337ec09ccaaa1374713aad9ba26cdc1308586a27cb309a883a0b19bde663d8R7229-R7236

because as of now Vyper does not have an entry in [lib/linguist/languages.yml](https://github.com/github/linguist/blob/master/lib/linguist/languages.yml) ... I wonder how GitHub currently manages to apply Vyper syntax to `.vy` files?

Screenshot 2022-02-03 at 21 51 58

**Comment by chiedo** [comment]
I think I may be out of my depths with how linguist works. 🤔 @lildude I'm going to defer to you.

**Comment by davidhq** [comment]
@lildude tl/dr Vyper syntax seems to be working and the only thing that is making it work seems to be this line:

Screenshot 2022-02-03 at 19 39 39

Vyper does not have an entry in `lib/linguist/languages.yml` at the moment.

**Comment by Nixinova** [comment]
> I wonder how GitHub currently manages to apply Vyper syntax to `.vy` files?

Syntax highlighting has been manually applied in that repo: see https://github.com/t4sk/vyper-by-example/blob/master/.gitattributes

**Comment by Nixinova** [comment]
> this is still useful, isn't it?

So yes this PR is still useful for properly adding Vyper

**Comment by davidhq** [comment]
> > I wonder how GitHub currently manages to apply Vyper syntax to `.vy` files?
> 
> Syntax highlighting has been manually applied in that repo: see https://github.com/t4sk/vyper-by-example/blob/master/.gitattributes

ok, this explains a lot, thank you

> > this is still useful, isn't it?
> 
> So yes this PR is still useful for properly adding Vyper

I also think so, but this should probably not be merged:

https://github.com/github/linguist/pull/5749/commits/7b98773827b87f61fe3cb1ba3db3f955735aebfa#diff-fe7afb5c9c916e521401d3fcfb4277d5071798c3baf83baf11d6071742823584R1128-R1130

just this part more or less I believe

Screenshot 2022-02-03 at 21 52 25

Reference to my grammar which is not copied directly from Python is already here:
https://github.com/github/linguist/blob/master/grammars.yml#L122

not sure if anything else has to be added to connect `source.vyper` to the [new Vyper definition entry](https://github.com/github/linguist/pull/5749/commits/7b98773827b87f61fe3cb1ba3db3f955735aebfa#diff-9c337ec09ccaaa1374713aad9ba26cdc1308586a27cb309a883a0b19bde663d8R7229-R7236)

**Comment by davidhq** [comment]
> not sure if anything else has to be added to connect `source.vyper` to the [new Vyper definition entry](https://github.com/github/linguist/pull/5749/commits/7b98773827b87f61fe3cb1ba3db3f955735aebfa#diff-9c337ec09ccaaa1374713aad9ba26cdc1308586a27cb309a883a0b19bde663d8R7229-R7236)

I think the answer to this is:

```yaml
Vyper:
  type: programming
  tm_scope: source.vyper
  color: "#46cf44"
  extensions:
  - ".vy"
  ace_mode: text
  language_id: 1055641948
```

so the `tm_scope: source.vyper` instead of `tm_scope: none`

I'm speaking to myself, I know that it was obvious to devs on linguist :}

**Comment by davidhq** [comment]
And not sure about the best color (currently #46cf44) which is green:
Screenshot 2022-02-03 at 22 41 18

who usually chooses this?

Vyper has this logo:
vyper

**Comment by Nixinova** [comment]
> I also think so, but this should probably not be merged: just this part more or less I believe

If you want Vyper to have its own syntax highlighting, than the other stuff should stay. If it's exactly the same syntax as Python, than the tm_scope can be changed to source.python & all the other stuff could be removed.

> And not sure about the best color (currently `#46cf44`) which is green: who usually chooses this?

The creator of the PR , with justification.
The [docs](https://vyper.readthedocs.io/en/latest/toctree.html) use blue `#2980b9` so maybe thats a better choice.

**Comment by Nixinova** [comment]
(Also, is this PR still 3 additions like the title says it is, or is it just Vyper now? If so the title should be changed.)

**Comment by Alhadis** [comment]
*Nota bene:* Ideally, this should've been submitted as three separate PRs, rather than a single monolithic PR.

**Comment by davidhq** [comment]
Hi!

@chiedo is it a good idea to maybe close this one:

https://github.com/github/linguist/issues/5741

and rename this current issue to **Add Vyper** ?

also to close these two or add relevant comments?

https://github.com/github/linguist/issues/5740
https://github.com/github/linguist/issues/5739

this would separate concerns as @Alhadis is suggesting.

then if anyone needs any of my further input on Vyper, I'm available... you can use my repo https://github.com/davidhq/SublimeEthereum/tree/linguist for this (it already has an entry in `grammars.yml` plus it's already used for Solidity) and if you plan to do it, I can invest more time into improving / double checking the quality of highlighter until next linguist release and additionaly for one after this after highlighter is observed in live environment.

**Comment by chiedo** [comment]
@davidhq feel free to take this and run with it! Through this PR, I've realized I don't have a strong enough understanding of Linguist.

So y'all can take it from here!

I'll close my issues also. Feel free to reference them in any new issues or PRs.

**Comment by davidhq** [comment]
@chiedo can you reopen this issue then and change it's name to **Add vyper**.

Then we wait for others to take it from here, I can't do much until vyper syntax from my repo is accepted, so I'll wait for that first.

**Comment by chiedo** [comment]
Done @davidhq!

**Comment by davidhq** [comment]
@chiedo tnx

@lildude and others, if I can help with preparing a PR based on info here and take a second look at vyper grammar inside my repo etc., let me know... this is provided that Vyper has now reached enough usage to merit inclusion... so if you provide input / opinion whenever suitable, it would be useful... thank you

**Comment by chiedo** [comment]
Thanks for your work!

**Pull request closed without merging**
